library.py

```diff
@@ -43,46 +37,69 @@ def add_book(filename, isbn, title, author):
     # to the books object.
 
     books[isbn] = {'title': title, 'author':author}
-
+    
     # Finally, write code that writes the new data to the library
     # Do we need to return anything?
     with open(filename) as f:
         json.dump({'students':students,'books':books}, f)
-
+        
+    
 def remove_book(filename, isbn):
     students, books = open_library(filename)
 
     # How can we *remove* an item from a dictionary?
     # Write code to delete the book keyed by isbn in the space below
 
+    if 'isbn' in books:
+        del books['isbn']
     # Now write code that saves the new version of the data to your library
-
+    with open(filename) as f:
+        json.dump ({'students' : students, 'books' : books} ,f)
+        
 def check_out(filename, isbn, s_id):
     students, books = open_library(filename)
 
     # Find a way to mark a book as checked out. Be sure to associate
     # the book with the student who borrowed it!
-
+    
+    checked_out = {}
```

Comment:
What happens if you run this function twice? It seems to me every time the function is run, the checked_out dictionary is reset.